libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -261,32 +265,26 @@ public void onUserOffRoute(Location location, boolean userOffRoute) {
 
   public void rerouteOccurred() {
     mapboxNavigation.setSessionState(mapboxNavigation.getSessionState().toBuilder()
+      .routeProgressBeforeReroute(routeProgress)
       .beforeRerouteLocations(Arrays.asList(
         locationBuffer.toArray(new Location[locationBuffer.size()])))
```

Comment:
Won't this set ~40 beforeRerouteLocations?

Reply:
No, Originally, I set this up similar to how iOS was handling this but I found that filtering based on time would be less efficient than simply storing an array of before and after location updates like we were previously doing. Essentially, when a reroute event occurs we capture a "snapshot" of the session state and hold onto it for ~20 seconds, add the after reroute locations and then push the event.